conformance/lang/expressions/is-expr/is_expr.balt

```diff
@@ -0,0 +1,1931 @@
+    io:println(error("message", error("AnotherErr"), code = 3, no = 1) is ErrorType4); // @output true
+    io:println(error("message", error("AnotherErr"), code = 3) is ErrorType5); // @output false
+}
+
+Test-Case: output
+Description: Test anonymous function expression as expression in type test expression.
+Labels: explicit-anonymous-function-expr, function-type, int, is-expr, string, union-type
+
+function init() {
+    io:println(function(string x, string y) returns (string) {
+        return x + y;
+    } is function); // @output true
+
+    io:println(function(string x, string y) returns (string) {
+        return x + y;
+    } is function (string, string) returns (string)); // @output true
+
+    io:println(function(string x, string y) returns (string) {
+        return x + y;
+    } is function (string, string|int) returns (string)); // @output false
+
+    io:println(function(string x, string y) returns (string) {
+        return x + y;
+    } is function (string, string) returns (string|int)); // @output true
+}
+
+Test-Case: output
+Description: Test let expression as expression in type test expression.
+Labels: decimal, float, int, is-expr, let-expr, list-constructor-expr, map-type, mapping-constructor-expr,
+        member-access-expr, module-const-decl, string, union-type
+
+const ONE = 1;
+
+function init() {
+    io:println(let int i = 2, float f = 2.2f, decimal d = 1.2d in <decimal>i + <decimal>f + d is decimal); // @output true
+    io:println(let var l = ["A", 2] in l[0] is string); // @output true
+    io:println(let map<int|string> m = {a: 1, b: "A"} in m["a"] is int); // @output true
+    io:println(let map<int|string> m = {a: 1, b: "A"} in m["a"] is ONE); // @output true
+}
+
+Test-Case: output
+Description: Test type cast expression as expression in type test expression.
+Labels: decimal, int, is-expr, module-const-decl, string, type-cast-expr, union-type
+
+const ONE = 1;
+
+type Type int|float;
+
+function init() {
+    io:println(<decimal>2 is decimal); // @output true
+
+    int|string a = 1;
+    io:println(<int>a is ONE); // @output true
+    io:println(<int>a is Type); // @output true
+}
+
+Test-Case: output
+Description: Test type of expression as expression in type test expression.
+Labels: int, is-expr, module-const-decl, string, typeof-expr, union-type
+
+const ONE = 1;
+
+function init() {
+    io:println(typeof 2 is typedesc<int>); // @output true
+    io:println(typeof 2 is typedesc<int|string>); // @output true
+
+    int|string a = 1;
+    io:println(typeof a is typedesc<ONE>); // @output true
+    io:println(typeof a is typedesc<int>); // @output true
+}
+
+Test-Case: output
+Description: Test unary logical expression as expression in type test expression.
+Labels: boolean, equality, int, is-expr, logical-expr, module-const-decl, relational-expr, string, union-type
+
+const TRUE = true;
+
+function init() {
+    io:println(!true is boolean); // @output true
```

Comment:
Shouldn't we get `always true` warning for `true is true` as we do in `1 is int`

Reply:
It seems we are not logging this warning when we have a literal as the expression and the finite type in the type descriptor.
Created https://github.com/ballerina-platform/ballerina-lang/issues/35559